### PR TITLE
Fix handling for "bundle" and "test-jar" reactor deps

### PR DESCRIPTION
**Pull request by Kortanul**
This fixes a few edge cases that were not covered by https://github.com/s4u/pgpverify-maven-plugin/pull/32.

This changes the skip filter for reactor dependencies to only compare based on Maven coordinates rather than the full ID of each artifact. Previously, the group ID, artifact ID, version, packaging, type, and classifier had to match, but unfortunately Maven doesn't have packaging information when resolving upstream reactor dependencies. This caused problems for `bundle` and `test-jar` packaging (which both end up being `jar` packaging under the hood). The new approach just uses coordinates of group ID, artifact ID, and version; this should work fine since it's unlikely that one would want a dependency outside the reactor that exactly matches the coordinates of a project inside the reactor during the same build.

### CODE REVIEW

File: src/main/java/org/simplify4u/plugins/skipfilters/ReactorDependencySkipper.java [hunk 1]
```diff
@@ -46,16 +56,58 @@ public boolean shouldSkipArtifact(Artifact artifact) {
...
+     * @param artifact1
+     *   The first artifact.
+     * @param artifact2
+     *   The second artifact.
+     *
+     * @return
+     *   If either the two artifacts are equal or correspond to the same Maven
+     *   coordinates.
+     */
+    @SuppressWarnings("PMD.UselessParentheses")
```
Comment (slawekjaranowski) [inline src/main/java/org/simplify4u/plugins/skipfilters/ReactorDependencySkipper.java]: is it need?

Comment (Kortanul) [reply src/main/java/org/simplify4u/plugins/skipfilters/ReactorDependencySkipper.java]: yes. PMD flagged the fact that I'm grouping the AND conditions with parens separately from the single OR on line 109 as an "unnecessary use of parens". I get the fact that the precedence order of `&&` is higher than `||`, but I wanted to be explicit that we're going to try the `.equals()` check first, and then if that does not pass, move on to comparing group, artifact ID, and version as the next logical group.
